packages/@aws-cdk/aws-codepipeline-actions/lib/codebuild/build-action.ts

```diff
@@ -139,8 +139,15 @@ export class CodeBuildAction extends Action {
       resources: [this.props.project.projectArn],
       actions: [
         'codebuild:BatchGetBuilds',
-        'codebuild:StartBuild',
-        'codebuild:StopBuild',
```

Comment:
Instead of a separate statement, let's just add this condition inline:

```suggestion
        `codebuild:${this.props.executeBatchBuild ? 'StartBuildBatch' : 'StartBuild'}`,
        `codebuild:${this.props.executeBatchBuild ? 'StopBuildBatch' : 'StopBuild'}`,
```

This way, you won't have to change any expectations in any existing tests!

Reply:
Great will do. It means `BatchGetBuilds` can appear twice in the cloudformation if you have a batch and none batch stage but I guess that’s ok.

Reply:
Yes, that's totally fine - those different stages should be using different Roles anyway.